blogposts/2022/introducting-migrator-service.md

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Introducing the migrator service'
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2022-02-15T12:00-00:00
+tags: [blog]
+slug: introducing-migrator-service
+heroImage: https://sourcegraphstatic.com/blog/introducing-migrator-service/introducing-migrator-service.jpg
+socialImage: https://sourcegraphstatic.com/blog/introducing-migrator-service/introducing-migrator-service.jpg
+published: true
+description: "We have identified and solved a huge source of pain during upgrades for Sourcegraph administrators. Here's how."
+---
+
+We have recently identified the root cause of a very frequent and insidious pain related to database migrations during Sourcegraph version upgrades. Many users handling the installation and upgrade of their Sourcegraph instance can likely attest to seeing an error similar to the following mid-way through an upgrade from one minor release to the next.
+
+```log
+ERROR: Failed to migrate the DB. Please contact support@sourcegraph.com for further assistance: Dirty database version 1528395797. Fix and force version.
+```
+
+This error message is the bane of any engineer in our customer support org, any engineer in our #opsgenie Slack channel, and a growing number of site administrators.
+
+As of Sourcegraph 3.37, we no longer spuriously mark the database as dirty during common upgrade scenarios. We have decoupled database migrations from the application startup sequence to better control database schema upgrade conditions.
+
+This release brings:
+
+- [New tools](https://docs.sourcegraph.com/admin/how-to/manual_database_migrations) that allow site administrators to run database migrations independently from an upgrade. Large instance upgrades can be upgraded ahead-of-time of an upgrade to reduce the possibility of downtime due to migrations taking longer than the healthcheck timeout.
```

Comment:
```suggestion
- [New tools](https://docs.sourcegraph.com/admin/how-to/manual_database_migrations) that allow site administrators to run database migrations independently from an upgrade. You can now make large instance upgrades before a general upgrade to reduce the possibility of downtime due to migrations taking longer than the healthcheck timeout.
```
Is "general" the right word here? Let me know if I've missed your meaning.

Reply:
```suggestion
- [New tools](https://docs.sourcegraph.com/admin/how-to/manual_database_migrations) that allow site administrators to run database migrations independently from an upgrade. Large database instances can be upgraded ahead-of-time of a general application upgrade to reduce the possibility of downtime due to migrations taking longer than the healthcheck timeout.
```

Reply:
@nickmyyz This better?